lua/telescope/actions/set.lua

```diff
@@ -196,7 +196,11 @@ action_set.edit = function(prompt_bufnr, command)
     -- check if we didn't pick a different buffer
     -- prevents restarting lsp server
     if vim.api.nvim_buf_get_name(0) ~= filename or command ~= "edit" then
-      filename = Path:new(filename):normalize(vim.loop.cwd())
+      if not vim.startswith(filename, "../") then
```

Comment:
If `filename` here is coming from `entry.path` and `entry.path` is a fully qualified path, then it can never start with `../`.

For this change to be necessary, I feel that something wrong has already happened in the entry maker.

Reply:
@jamestrew 
Here is an example -
1. `cwd = /home/d/code/myproject`
2. `linux` source is located at `/home/d/code/linux`
3. I have configured cscope db in both `myproject` and `linux`
4. When query is made for definition of some `symbol` from `myproject`, I get results from `linux` directory.
5. This result is something like `/home/d/code/linux/some/header/file.h` (there could be more results)
6. Instead of passing this abs path, I pass `../linux/some/header/file.h` and avoid seeing my home directory repeatedly.
7. This path is "sanitized" by telescope making it `/linux/some/header/file.h`
8. Now this is invalid path.

Reply:
I understand the use case. You're free to display a relative path (in the `entry.display` function). You just need to pass an absolute path for `entry.path`.